### ISSUE TITLE
Add support for (optional) parallel test execution

### ISSUE THREAD

**Issue by compnerd** [swift-server/vscode-swift#640]
**Is your feature request related to a problem? Please describe.**
Currently the test execution is serial.  This makes executing large test suites annoyingly long.

**Describe the solution you'd like**
Having the option to flip tests to parallel would reduce the execution time.  This should be optional as I have seen some cases where the serial execution will reproduce a problem that parallel execution does not. 

**Describe alternatives you've considered**
Continuing to run the tests sequentially is possible, but does not help developer productivity.

**Additional context**

**Comment by tristanlabelle** [comment]
It seems like this is a matter of adding `--parallel` to either

https://github.com/swift-server/vscode-swift/blob/20e0907360cd891d3de72526d4c0084d7bc86414/src/debugger/launch.ts#L177

Or probably better:

https://github.com/swift-server/vscode-swift/blob/20e0907360cd891d3de72526d4c0084d7bc86414/src/TestExplorer/TestRunner.ts#L285C18-L285C18

And maybe having an extension config to control it.

**Comment by adam-fowler** [comment]
Parallel testing output isn't parsable. You can't tell if tests have passed, failed or skipped easily.

There are some future plans for SwiftPM which include a parsable test output. Maybe that will work with parallel testing. 

This is also a duplicate

**Comment by KrisSimon** [comment]
> Parallel testing output isn't parsable. You can't tell if tests have passed, failed or skipped easily.

Hi Adam, 
sorry if this is a stupid question, I am not deep into the internals, but what is about writing the output to `--xunit-output` and parsing this one.  
this is how we do it in our CI for projects, maybe xunit output could be the parsable input for the results browser. 

To run test with --parallel would be great!

**Comment by adam-fowler** [comment]
I didn't know about the xUnit output. It probably could be used in certain situations. From what I can tell SwiftPM parallel testing is controlled by SwiftPM and not XCTest. In the situation where I want to debug a test process I have to run XCTest directly. If I use `swift test` the debugger attaches to the swift package manager process which isn't much use. 

Also the last time I checked Windows doesn't like me calling `swift test` I have to call the `xctest` executable.  This is why code coverage does not work on Windows, as I need to use `swift test --enable-code-coverage` to get coverage results.

So if we are not debugging the test process and we aren't on Windows we could possibly use the xUnit output from the parallel tests.

I've duplicated this text into the open bug related to parallel testing #433
